src/sqlancer/materialize/MaterializeGlobalState.java

```diff
@@ -0,0 +1,292 @@
+package sqlancer.materialize;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import sqlancer.Randomly;
+import sqlancer.SQLConnection;
+import sqlancer.SQLGlobalState;
+
+public class MaterializeGlobalState extends SQLGlobalState<MaterializeOptions, MaterializeSchema> {
+
+    public static final char IMMUTABLE = 'i';
+    public static final char STABLE = 's';
+    public static final char VOLATILE = 'v';
+
+    private List<String> operators = Collections.emptyList();
+    private List<String> collates = Collections.emptyList();
+    private List<String> opClasses = Collections.emptyList();
+    private List<String> tableAccessMethods = Collections.emptyList();
+    // store and allow filtering by function volatility classifications
+    private final Map<String, Character> functionsAndTypes = new HashMap<>();
+    private List<Character> allowedFunctionTypes = Arrays.asList(IMMUTABLE, STABLE, VOLATILE);
+
+    @Override
+    public void setConnection(SQLConnection con) {
+        super.setConnection(con);
+        try {
+            this.opClasses = getOpclasses();
+            this.operators = getOperators(getConnection());
+            this.collates = getCollnames(getConnection());
+            this.tableAccessMethods = getTableAccessMethods(getConnection());
+        } catch (SQLException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    private List<String> getCollnames(SQLConnection con) throws SQLException {
+        List<String> collNames = new ArrayList<>();
+        try (Statement s = con.createStatement()) {
+            try (ResultSet rs = s
+                    .executeQuery("SELECT collname FROM pg_collation WHERE collname LIKE '%utf8' or collname = 'C';")) {
+                while (rs.next()) {
+                    collNames.add(rs.getString(1));
+                }
+            }
+        }
+        return collNames;
+    }
+
+    private List<String> getOpclasses() throws SQLException {
+        List<String> opClasses = new ArrayList<>();
+        // select opcname FROM pg_opclass;
```

Comment:
In the future, one option would be to query this information from the database system directly to reduce maintenance effort (I think no SQLancer implementations do this yet).